bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonDeviceHandler.java

```diff
@@ -322,18 +333,30 @@ public void channelLinked(ChannelUID channelUID) {
                 new InsteonAddress(config.getAddress()), productKey, params);
         getInsteonBinding().addFeatureListener(bindingConfig);
 
-        logger.debug("channel {} linked with the feature: {} parameters: {}", channelUID.getAsString(), feature,
-                params);
+        StringBuilder builder = new StringBuilder(channelUID.getAsString());
+        builder.append(" feature = ");
+        builder.append(feature);
+        builder.append(" parameters = ");
+        builder.append(params);
+        String msg = builder.toString();
+        logger.debug("{}", msg);
+
+        getInsteonNetworkHandler().linked(channelUID, msg);
     }
 
     @Override
     public void channelUnlinked(ChannelUID channelUID) {
         getInsteonBinding().removeFeatureListener(channelUID);
+        getInsteonNetworkHandler().unlinked(channelUID);
 
         logger.debug("channel {} unlinked ", channelUID.getAsString());
     }
 
+    private @Nullable InsteonNetworkHandler getInsteonNetworkHandler() {
+        return (InsteonNetworkHandler) getBridge().getHandler();
+    }
+
     private @Nullable InsteonBinding getInsteonBinding() {
-        return ((InsteonNetworkHandler) getBridge().getHandler()).getInsteonBinding();
+        return getInsteonNetworkHandler().getInsteonBinding();
```

Comment:
Getting the InsteonNetworkHandler from the bridge is preferred to getting it statically.

Reply:
It does, just eliminating some redundant code: 

```
private @Nullable InsteonNetworkHandler getInsteonNetworkHandler() {
    return (InsteonNetworkHandler) getBridge().getHandler();
}
```

Reply:
getInsteonNetworkHandler() does get it from the bridge. It right above this method.

Reply:
oops, I glanced right over it and didn't see it.